tests/api/test_addon_uploads.py

```diff
@@ -1922,6 +1953,285 @@ def test_edit_version_both_license_and_custom_license(base_url, session_auth):
+def test_extension_valid_compatibility_values(
+    base_url, session_auth, request_value, response_value
+):
+    """Tests the compatibility field with a set of valid values"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    payload = {**payloads.edit_version_details, 'compatibility': request_value}
+    edit_version = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    print(
+        f'For compatibility "{request_value}": Response status is '
+        f'{edit_version.status_code}; {edit_version.text}\n'
+    )
+    r = edit_version.json()
+    assert (
+        edit_version.status_code == 200
+    ), f'Actual status code was {edit_version.status_code}'
+    # verify that the returned compatibility matches what we expect for the values sent
+    assert response_value == r['compatibility']
+
+
+@pytest.mark.parametrize(
+    'value',
+    [
+        [None, None],
+        ['firefox', None],
+        {'firefox': {'min': '*'}},
+        {'firefox': {'min': '78.*'}},
+        {'android': {'max': '99595.0'}},
+        {'firefox': {'min': '*', 'max': '65.0'}},
+    ],
+    ids=[
+        'None/No values',
+        'Valid app for firefox, None for android',
+        'Invalid firefox min appversion - "*"',
+        'Invalid firefox min appversion - "x.*"',
+        'Unavailable max appversion',
+        'Valid max version, invalid min version',
+    ],
+)
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_edit_version_invalid_compatibility_values(base_url, session_auth, value):
+    """Compatibility values should be a combination of valid applications (firefox or android)
+    and application versions (existing versions of Firefox for desktop/android)"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    payload = {**payloads.edit_version_details, 'compatibility': value}
+    edit_version = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    print(
+        f'For compatibility "{value}": Response status is '
+        f'{edit_version.status_code}; {edit_version.text}\n'
```

Comment:
Is this supposed to be here?

Reply:
I left the print there because it helps me debug failures more easily by checking the response code and the response message before the actual validations (assertions).
I could add all of this in the first assert message (the one immediately following) as an alternative solution.

Reply:
Well if you think it's useful I don't mind, I do wonder if the assert message idea would be a bit cleaner. Up to you though.